tracer/src/Datadog.Trace/SpanContextInjector.cs

```diff
@@ -0,0 +1,37 @@
+// <copyright file="SpanContextInjector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.Telemetry.Metrics;
+
+#nullable enable
+
+namespace Datadog.Trace
+{
+    /// <inheritdoc />
+    public class SpanContextInjector : ISpanContextInjector
```

Comment:
FYI this is public surface change, so you'll need to update the snapshots.

I wonder if the `ISpanContextInjector` adds any value here. Remember, we can't evolve interfaces (as they're breaking changes), and I don't think it serves any purpose tbh, as the customer needs to create an instance of the `SpanContextInjector` anyway 🤔

Reply:
Ok, added it for mocking but ok to remove it. Done in c2b32c35c146f3522d6fabe4631c6b825d543a8e